libs/designsystem/src/lib/components/popover/popover.component.ts

```diff
@@ -0,0 +1,210 @@
+import { DOCUMENT } from '@angular/common';
+import {
+  AfterViewInit,
+  Component,
+  ElementRef,
+  EventEmitter,
+  HostListener,
+  Inject,
+  Input,
+  OnChanges,
+  OnDestroy,
+  Output,
+  Renderer2,
+  SimpleChanges,
+  ViewChild,
+} from '@angular/core';
+
+import { DesignTokenHelper } from '../../helpers';
+
+export enum HorizontalDirection {
+  right = 'right',
+  left = 'left',
+}
+@Component({
+  selector: 'kirby-popover',
+  template: `
+    <div #wrapper class="wrapper"><ng-content></ng-content></div>
+  `,
+  styleUrls: ['./popover.component.scss'],
+})
+export class PopoverComponent implements AfterViewInit, OnChanges, OnDestroy {
+  @ViewChild('wrapper', { static: true, read: ElementRef })
+  wrapperElement: ElementRef<HTMLDivElement>;
+
+  @Input()
+  target: HTMLElement | ElementRef<HTMLElement>;
+
+  @Output()
+  willHide = new EventEmitter<void>();
+
+  private targetElement: HTMLElement;
```

Comment:
This is more or less a computed property of target - consider turning it into a getter function instead to not have to maintain two pieces of state.